examples/pytorch_ignite/gin.py

```diff
@@ -0,0 +1,214 @@
+import os.path as osp
+
+import ignite
+import ignite.contrib.handlers.tensorboard_logger
+import ignite.contrib.handlers.tqdm_logger
+import torch
+
+import torch_geometric as pyg
+
+
+class Model(torch.nn.Module):
+    def __init__(self, in_channels: int, out_channels: int,
+                 hidden_channels: int = 64, num_layers: int = 3,
+                 dropout: float = 0.5):
+        super().__init__()
+        self.gnn = pyg.nn.GIN(in_channels, hidden_channels, num_layers,
+                              dropout=dropout, jk='cat')
+
+        self.classifier = pyg.nn.MLP(
+            [hidden_channels, hidden_channels, out_channels], batch_norm=True,
+            dropout=dropout)
+
+    def forward(self, data):
+        x, edge_index, batch = data.x, data.edge_index, data.batch
+        x = self.gnn(x, edge_index)
+        x = pyg.nn.global_add_pool(x, batch)
+        x = self.classifier(x)
+        return x
+
+
+def main():
+    pyg.seed_everything(42)
+
+    device = torch.device('cuda')
+    root = osp.join('data', 'TUDataset')
+    log_dir = 'runs/example/'
+    dataset = pyg.datasets.TUDataset(
+        root, 'IMDB-BINARY', pre_transform=pyg.transforms.OneHotDegree(135))
+
+    amp_mode = 'amp'
+    batch_size = 16
+
+    dataset = dataset.shuffle()
+    test_dataset = dataset[:len(dataset) // 10]
+    val_dataset = dataset[len(dataset) // 10:2 * len(dataset) // 10]
+    train_dataset = dataset[2 * len(dataset) // 10:]
+    loader_trn = pyg.data.DataLoader(
+        dataset=train_dataset,
+        batch_size=batch_size,
+        shuffle=True,
+        num_workers=1,
```

Comment:
```suggestion
        num_workers=1,
```
I think we either want to increase `num_workers` or set it to `0`. In my benchmarks, `num_workers=1` performs quite poorly.